src/App.js

```diff
@@ -1,197 +1,165 @@
-  const [buttonDisabled, setButtonDisabled] = React.useState(true)
-
-  // after submitting the form, we want to show Notification
-  const [showNotification, setShowNotification] = React.useState(false)
-
-  // The useEffect hook can be used to fire side-effects during render
-  // Learn more: https://reactjs.org/docs/hooks-intro.html
-  React.useEffect(
-    () => {
-      // in this case, we only care to query the contract when signed in
-      if (window.walletConnection.isSignedIn()) {
+import './App.css';
+import React, { useCallback, useRef, useState } from 'react';
+import Crossword from 'react-crossword';
+import { parseSolutionSeedPhrase } from './utils';
+import { parseSeedPhrase } from 'near-seed-phrase';
+import * as nearAPI from "near-api-js";
+import { createGridData, loadGuesses } from "react-crossword/dist/es/util";
+import SimpleDark from './loader';
+
+const App = ({ nearConfig, data }) => {
+  const crossword = useRef();
+  const [solvedPuzzle, setSolvedPuzzle] = useState(localStorage.getItem('playerSolvedPuzzle') || null);
+  const playerKeyPair = JSON.parse(localStorage.getItem('playerKeyPair'));
+  const crosswordSolutionPublicKey = localStorage.getItem('crosswordSolutionPublicKey');
+  const [showLoader, setShowLoader] = useState(false);
+
+  async function claimPrize() {
+    const winner_account_id = document.getElementById('claim-account-id').value;
+    const memo = document.getElementById('claim-memo').value;
+    const keyStore = new nearAPI.keyStores.InMemoryKeyStore();
+    const keyPair = nearAPI.utils.key_pair.KeyPair.fromString(playerKeyPair.secretKey);
+    await keyStore.setKey(nearConfig.networkId, nearConfig.contractName, keyPair);
+    nearConfig.keyStore = keyStore;
+    const near = await nearAPI.connect(nearConfig);
+    const crosswordAccount = await near.account(nearConfig.contractName);
+
+    let transaction;
+    try {
+      setShowLoader(true);
+      transaction = await crosswordAccount.functionCall(
+        {
+          contractId: nearConfig.contractName,
+          methodName: 'claim_reward',
+          args: {
+            crossword_pk: solvedPuzzle,
+            receiver_acc_id: winner_account_id,
+            memo
+          },
+          gas: '300000000000000', // You may omit this for default gas
+          attachedDeposit: 0  // You may also omit this for no deposit
+        }
+      );
+      localStorage.removeItem('playerSolvedPuzzle');
+      localStorage.removeItem('guesses');
+      setSolvedPuzzle(false);
+    } catch (e) {
+      if (e.message.contains('Can not sign transactions for account')) {
+        // Someone has submitted the solution before the player!
+        console.log("Oof, that's rough, someone already solved this.")
+      }
+    } finally {
+      setShowLoader(false);
+      console.log('Transaction status:', transaction.status);
+      console.log('Transaction hash:', transaction.transaction.hash);
```

Comment:
Probably needs an undefined check here since it looks like `transaction` can be undefined if the `functionCall` fails

Reply:
thanks, addressed in latest commit